doc/docs/json-schema.md

```diff
@@ -0,0 +1,98 @@
+# Generating JSON Schema
+
+Since Tapir v1.4.0 you can conveniently generate JSON schema from Tapir schema, which can be derived from your Scala types. Use `TapirSchemaToJsonSchema`:
+
+```scala
+"com.softwaremill.sttp.tapir" %% "tapir-apispecs-docs" % "@VERSION@"
+// if you need to serialize the schema. See https://github.com/softwaremill/sttp-apispec
+"com.softwaremill.sttp.apispec" %% "jsonschema-circe" % "..."
+```
+
+Schema generation can now be performed like in the following example:
+
+```scala:mdoc:compile-only
+import sttp.apispec.circe._
+import sttp.apispec.{ReferenceOr, Schema => ASchema, SchemaType => ASchemaType}
+import sttp.tapir._
+import sttp.tapir.docs.apispec.schema._
+import sttp.tapir.generic.auto._
+
+  object Childhood {
+    case class Child(age: Int, height: Option[Int])
+  }
+  case class Parent(innerChildField: Child, childDetails: Childhood.Child)
+  case class Child(childName: String) // to illustrate unique name generation
+  val tSchema = implicitly[Schema[Parent]]
+
+  val jsonSchema: ReferenceOr[ASchema] = TapirSchemaToJsonSchema(
+    tSchema,
+    markOptionsAsNullable = true,
+    metaSchema = MetaSchemaDraft04, // default
+    schemaName = defaultSchemaName // default
+)
+```
+
+All the nested schemas will be referenced from the `$defs` element.
+In order to generate a JSON representation of the schema, you can use Circe:
```

Comment:
that would be a good place to mention the extra dependency that's needed

Reply:
It's mentioned here https://github.com/softwaremill/tapir/pull/2904/files#diff-a2236d1f0584452eefe52cb190b04ffdf887407b7237a121fb838c3df9017325R8
Do you think we should repeat that here?

Reply:
Ok, I rearranged the document to divide more clearly between conversion and serialization, including imports. I also added some more fixes.